### ISSUE TITLE
fedora-toolbox create fails to generate new container if image already exists

### ISSUE THREAD

**Issue by jeslinmx** [debarshiray/fedora-toolbox#19]
tl;dr - declare dbus_system_bus_path outside of the `buildah inspect --type image $toolbox_image` check

Hi there!

After messing up my pet container, I decided to start afresh by running:
```
[jeshua@jeshua-Surface-Pro-3 ~]$ podman stop fedora-toolbox-jeshua:29 
d3ac6a0f1b80dea6f6fd64204ffd8a444ee4bccb7a16d797540984e48357916e
[jeshua@jeshua-Surface-Pro-3 ~]$ podman rm fedora-toolbox-jeshua:29 
d3ac6a0f1b80dea6f6fd64204ffd8a444ee4bccb7a16d797540984e48357916e
```
If I'm not mistaken, fedora-toolbox create is expected to consider that while the container is gone, the image is still there, and pick up the process from there, generating a fresh container from the image. However, that step fails:
```
[jeshua@jeshua-Surface-Pro-3 ~]$ fedora-toolbox create
/home/jeshua/.local/bin/fedora-toolbox: failed to create container fedora-toolbox-jeshua:29
```
Considering the verbose output,
```
error looking up container "fedora-toolbox-jeshua:29": no container with name or ID fedora-toolbox-jeshua:29 found: no such container
invalid host path, must be an absolute path ""
/home/jeshua/.local/bin/fedora-toolbox: failed to create container fedora-toolbox-jeshua:29
```
the first line is as expected (its from the check on whether the container exists), but the second is causing podman create to fail. A bunch of debug echos reveals that --volume $dbus_system_bus_path:$dbus_system_bus_path is to blame, since that variable is declared inside of the block that only runs if the image needs to be created.

Recommendation: move
```
if ! $prefix_sudo buildah inspect --type image $toolbox_image >/dev/null 2>&42; then
        ...
        if [ "$DBUS_SYSTEM_BUS_ADDRESS" != "" ]; then
            dbus_system_bus_address=$DBUS_SYSTEM_BUS_ADDRESS
        fi
        dbus_system_bus_path=$(echo $dbus_system_bus_address | cut --delimiter = --fields 2 2>&42)
        dbus_system_bus_path=$(readlink --canonicalize $dbus_system_bus_path 2>&42)
        ...
fi
```
out of the surrounding conditional.

Sorry if this is a bit lengthy for what amounts to a tiny fix, first time reporting an issue.

**Comment by juhp** [comment]
Thanks for this

However for me fedora-toolbox is trying to pull fedora-toolbox-<USER> somehow! - I deleted all the toolbox containers and images (this is after having had a successful fedora-toolbox 28 container created earlier and running earlier)

**Comment by debarshiray** [comment]
Thanks @jeslinmx for the accurate description and analysis. This was fixed by @mvyletel in #22.
